util/personal_names.py

```diff
@@ -33,6 +39,18 @@ def replacePhD(match):
     return match.groups()[0] + "PhD"
 
 
+def replaceEndPunctuation(match):
+    """
+    If there was found to be improper punctuation at the end of the name string, 
+    clean it off.
+    :param match: a regular expression matched to a string
+    """
+    if not match or len(match.groups()) < 3:
+        return match
+
+    return match.groups()[0] + match.groups()[1]
+
+
```

Comment:
Sorry to nitpick but there are a couple problems with this function. First, it should be called replace_end_punctuation for PEP 8 compliance. Second, it's designed for use in one specific place -- inside `name_tidy` -- so the best thing to do is to define it within that function, just before it's used. Otherwise it looks like a general name cleanup utility function that can be called by whoever.